plugins/modules/proxmox_kvm.py

```diff
@@ -1018,23 +1038,24 @@ def create_vm(self, vmid, newid, node, name, memory, cpu, cores, sockets, update
             urlencoded_ssh_keys = quote(kwargs['sshkeys'], safe='')
             kwargs['sshkeys'] = str(urlencoded_ssh_keys)
 
-        # If update, don't update disk (virtio, efidisk0, tpmstate0, ide, sata, scsi) and network interface
+        # If update, don't update disk (virtio, efidisk0, tpmstate0, ide, sata, scsi) and network interface, unless update_unsafe=True
         # pool parameter not supported by qemu/<vmid>/config endpoint on "update" (PVE 6.2) - only with "create"
         if update:
-            if 'virtio' in kwargs:
-                del kwargs['virtio']
-            if 'sata' in kwargs:
-                del kwargs['sata']
-            if 'scsi' in kwargs:
-                del kwargs['scsi']
-            if 'ide' in kwargs:
-                del kwargs['ide']
-            if 'efidisk0' in kwargs:
-                del kwargs['efidisk0']
-            if 'tpmstate0' in kwargs:
-                del kwargs['tpmstate0']
-            if 'net' in kwargs:
-                del kwargs['net']
+            if update_unsafe is not True:
```

Comment:
```suggestion
            if update_unsafe is False:
```
or
```suggestion
            if not update_unsafe:
```
makes this easier to understand.